### PR TITLE
[examples] fix sys.path in conftest.py

### PR DESCRIPTION
**Pull request by stas00**
the pt/tf reshuffle broke the examples tests setup. This PR fixes it, by 

* fixing `sys.path` setting now that it's one level down 

@sgugger

### CODE REVIEW

**Comment by sgugger** [comment]
That can't work since `conftest` now believes the module "tensorflow" is the folder in the examples with the same name, which is the reason why I moved it.

Path in conftest need to be adapted instead.

**Comment by stas00** [comment]
ah, OK, then I will make copies of it. But you can't move it since sys.path was relying on it being in a top-level subfolder - so will have to fix that or make the code more flexible. I will take care of it.

thank you for letting me know tf didn't like it.

**Comment by stas00** [comment]
ok, 

1. put `conftest.py` back and fixed it to set up `sys.path` correctly
2. made 2 copies of it for legacy/research.

File: examples/legacy/conftest.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    pytest_addoption_shared(parser)
+
+
+def pytest_terminal_summary(terminalreporter):
+    from transformers.testing_utils import pytest_terminal_summary_main
+
+    make_reports = terminalreporter.config.getoption("--make-reports")
+    if make_reports:
+        pytest_terminal_summary_main(terminalreporter, id=make_reports)
```
Comment (sgugger) [inline examples/legacy/conftest.py:44]: I don't understand the purpose of this file. legacy examples should not be tested anymore.

File: examples/research_projects/conftest.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+    pytest_addoption_shared(parser)
+
+
+def pytest_terminal_summary(terminalreporter):
+    from transformers.testing_utils import pytest_terminal_summary_main
+
+    make_reports = terminalreporter.config.getoption("--make-reports")
+    if make_reports:
+        pytest_terminal_summary_main(terminalreporter, id=make_reports)
```
Comment (sgugger) [inline examples/research_projects/conftest.py:44]: Same for this one, research projects should not be tested.

**Review by sgugger — APPROVED** [review]
The fix in itself LGTM but why add a conftest to legacy and research-projects? Both are not supposed to be tested.

**Comment by stas00** [comment]
> The fix in itself LGTM but why add a conftest to legacy and research-projects? Both are not supposed to be tested.

I'm adding deepspeed tests for wav2vec2 - it's been complicated to make it work (some parts still don't work). I don't think it's a good idea to put these tests under common tests.

So I can't run the tests w/o this conftest on dev box.

File: examples/legacy/conftest.py (hunk 1 above)
Comment (stas00) [reply examples/legacy/conftest.py]: I removed the reports part - you're correct - it's not needed.

just left the `sys.path` setup.

**Comment by sgugger** [comment]
I really don't think it's a good idea to add this to `research-projects`: all research projects have a pinned version of Transformers and this would use the master install to test them, which is incompatible with what is advertised.

Maybe add it specifically to the wav2vec2 research project? Or maybe wav2vec2 should move to a maintained example @patrickvonplaten ?

**Comment by stas00** [comment]
`pathlib` is easier to use for finding out older ancestors than doing dirname 4 times:
```
import sys
from pathlib import Path
git_repo_path = Path(__file__).resolve().parents[3] / "src"
sys.path.insert(1, str(git_repo_path))
```

**Pull request merged**